### PR TITLE
token-cli: error handling fixes

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
In several places in the cli `Result`s are unwrapped without error handling.

#### Summary of changes
Check `Result`s for `Err`.

### CODE REVIEW

File: token/cli/src/config.rs [hunk 1]
```diff
@@ -355,8 +362,11 @@ impl<'a> Config<'a> {
         address_name: &str,
         wallet_manager: &mut Option<Arc<RemoteWalletManager>>,
     ) -> Result<Pubkey, Error> {
-        if let Some(address) = pubkey_of_signer(arg_matches, address_name, wallet_manager).unwrap()
-        {
+        let address = match pubkey_of_signer(arg_matches, address_name, wallet_manager) {
+            Ok(address) => address,
+            Err(e) => return Err(e.to_string().into()),
+        };
```
Comment (joncinque) [inline token/cli/src/config.rs:368]: nit: how about bubbling this in one line by mapping the error?
```suggestion
        let address = pubkey_of_signer(arg_matches, address_name, wallet_manager).map_err(|e| e.to_string().into())?;
```

File: token/cli/src/config.rs [hunk 2]
```diff
@@ -326,13 +330,16 @@ impl<'a> Config<'a> {
         wallet_manager: &mut Option<Arc<RemoteWalletManager>>,
         token: Option<Pubkey>,
     ) -> Result<Pubkey, Error> {
-        if let Some(address) = pubkey_of_signer(arg_matches, override_name, wallet_manager).unwrap()
-        {
+        let address = match pubkey_of_signer(arg_matches, override_name, wallet_manager) {
+            Ok(address) => address,
+            Err(e) => return Err(e.to_string().into()),
+        };
```
Comment (joncinque) [inline token/cli/src/config.rs:336]: nit: how about mapping the error?
```suggestion
        let address = pubkey_of_signer(arg_matches, override_name, wallet_manager).map_err(|e| e.to_string().into())?;
```

File: token/cli/src/config.rs [hunk 3]
```diff
@@ -307,14 +307,18 @@ impl<'a> Config<'a> {
...
     ) -> Result<Pubkey, Error> {
-        let token = pubkey_of_signer(arg_matches, "token", wallet_manager).unwrap();
-        self.associated_token_address_for_token_or_override(
-            arg_matches,
-            override_name,
-            wallet_manager,
-            token,
-        )
-        .await
+        match pubkey_of_signer(arg_matches, "token", wallet_manager) {
```
Comment (joncinque) [inline token/cli/src/config.rs:310]: nit: we can keep this roughly the same by mapping the error, ie
```suggestion
        let token = pubkey_of_signer(arg_matches, "token", wallet_manager).map_err(|e| e.to_string().into())?;
```

**Review by joncinque — COMMENTED** [review]
Thanks for noticing these and contributing some fixes! Just a nit to map the error rather than matching

File: token/cli/src/config.rs (hunk 1 above)
Comment (jbiseda) [reply token/cli/src/config.rs]: good suggestion, updated all instances with `map_err`.

**Comment by jbiseda** [comment]
> Thanks for noticing these and contributing some fixes! Just a nit to map the error rather than matching

no problem, just happened to stumble across this

**Review by joncinque — APPROVED** [review]
The twoxtx job is getting fixed in #4125 so feel free to merge once CI passes

**Pull request merged**